staging/src/k8s.io/kube-aggregator/pkg/controllers/openapi/aggregator.go

```diff
@@ -52,12 +52,18 @@ type specAggregator struct {
 
 	// provided for dynamic OpenAPI spec
 	openAPIVersionedService *handler.OpenAPIService
+
+	// initialized blocks local spec installation after startup
+	initialized bool
 }
 
 var _ AggregationManager = &specAggregator{}
 
 // This function is not thread safe as it only being called on startup.
 func (s *specAggregator) addLocalSpec(spec *spec.Swagger, localHandler http.Handler, name, etag string) {
+	if s.initialized {
+		panic(fmt.Sprintf("local spec installation forbidden after startup: %v", name))
```

Comment:
a panic here is bad. At the very least an error. But why do we need this?

Reply:
it's a guard to prevent programming error. dropped